File: src/main/java/net/cydhra/nidhogg/requests/AuthenticateResponse.java
package net.cydhra.nidhogg.requests;

import lombok.Data;

import java.util.List;

/**
 *
 */
@Data
public class AuthenticateResponse {
    private final String accessToken;
    private final String clientToken;
    
    private final List<Profile> availableProfiles;
    private final Profile selectedProfile;
    private final User user;
    
    @Data
    public static class Profile {
        private final String id;
        private final String name;
        private final boolean legacy;
    }
    
    @Data
    public static class User {
        private final String id;
        private final List<UserProperty> properties;
    }
    
    @Data
    public static class UserProperty {
        private final String name;
        private final String value;
    }
}

File: src/main/java/net/cydhra/nidhogg/requests/AuthenticateResponse.kt
package net.cydhra.nidhogg.requests

internal class AuthenticateResponse(val accessToken: String,
                                    val clientToken: String,
                                    val availableProfiles: List<Profile>?,
                                    val selectedProfile: Profile?,
                                    val user: User?)

internal data class Profile(val id: String, val name: String, val isLegacy: Boolean)

internal data class User(val id: String, val properties: List<UserProperty>)

internal data class UserProperty(val value: String, val name: String)

